taxes/calculator.js

```diff
@@ -1,13 +1,29 @@
-module.exports = (productPrice, taxes) => {
-    const result = {...taxes}
-    Object.keys(taxes).forEach((tax) => {
-        if (typeof taxes[tax] === 'object') {
-            Object.keys(taxes[tax]).forEach((subTax) => {
-                result[tax][subTax] = productPrice * taxes[tax][subTax] / 100
-            });
-        } else {
-            result[tax] = productPrice * taxes[tax] / 100
-        }
-    })
-    return result;
-}
\ No newline at end of file
+class Calculator {
+    constructor(productPrice, state) {
+        this.productPrice = productPrice;
+        this.state = state;
+        this.cityTaxes = {
+            "NY": {vat: 23, specific: {city: 1, insurance: 2}, state: 5},
+            "AL": {vat: 30, specific: { cold_weather: 5 }, state: 0},
+        };
+        this.defaultTax = {vat: 21, state: 2};
```

Comment:
while it's ok to have the config inside constructor, it is relative strange to see here the arguments of calculate method